backend/wayland/seat.c

```diff
@@ -282,12 +283,40 @@ static struct wlr_wl_input_device *get_wl_input_device_from_input_device(
 	return (struct wlr_wl_input_device *)wlr_dev;
 }
 
+void create_wl_seat(struct wl_seat *wl_seat, struct wlr_wl_backend *wl) {
+	wl->seat.wl_seat = wl_seat;
+	wl_seat_add_listener(wl_seat, &seat_listener, wl);
+}
+
+void destroy_wl_seats(struct wlr_wl_backend *wl) {
+	if (wl->seat.pointer) {
+		wl_pointer_destroy(wl->seat.pointer);
+	}
+	free(wl->seat.name);
+	if (wl->seat.wl_seat) {
+		wl_seat_destroy(wl->seat.wl_seat);
+	}
+}
+
+static struct wlr_wl_seat *backend_get_seat(struct wlr_wl_backend *backend, struct wl_seat *wl_seat) {
+	assert(backend->seat.wl_seat == wl_seat);
+	return &backend->seat;
+}
+
+static struct wlr_wl_seat *input_device_get_seat(struct wlr_input_device *wlr_dev) {
+	struct wlr_wl_input_device *dev =
+		get_wl_input_device_from_input_device(wlr_dev);
+	// TODO: assert on pointers match (e.g. keyboard/pointer)?
```

Comment:
We should do this.

Reply:
Unfortunately at this moment we don't store any association between `wlr_input_device` and say `wl_keyboard` other than subscribing to it. I left this to-do at moment where I didn't grasped Wayland backend fully enough.